code/ui/generalhud/menu/Menu.cs

```diff
@@ -0,0 +1,95 @@
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+namespace TTTReborn.UI
+{
+    public class Menu : Panel
+    {
+        public static Menu Instance;
+
+        public bool IsShowing
+        {
+            get => _isShowing;
+            set
+            {
+                _isShowing = value;
+
+                SetClass("hide", !_isShowing);
+            }
+        }
+
+        private bool _isShowing = false;
```

Comment:
I wonder if we should have our own custom derived class that has this functionality, and each UI element can inherit from it, what do you think? We can add this as an issue.